app/models/trip.rb

```diff
@@ -0,0 +1,161 @@
+class Trip < ActiveRecord::Base
+
+  validates :duration,
+            :start_date,
+            :start_station_id,
+            :end_date,
+            :end_station_id,
+            :bike_id,
+            :subscription_type_id,
+            :user_zip_code,
+            :start_time,
+            :end_time,
+            presence: true
+
+  validates_uniqueness_of :bike_id, scope: [:start_date, :start_time]
+
+  belongs_to :start_station, class_name: 'Station', :foreign_key => :start_station_id
+  belongs_to :end_station, class_name: 'Station', :foreign_key => :end_station_id
+  belongs_to :subscription_type
+  belongs_to :weather_condition, :foreign_key => :start_date
+
+  ### Show Station methods ###
+  def self.number_rides_started_at_station(id)
+    where(start_station_id: id).count
+  end
+
+  def self.number_rides_ended_at_station(id)
+    where(end_station_id: id).count
+  end
+
+  def self.station_with_maximum_rides_as_start_location
+    station_counts = group(:start_station).count
+    station        = station_counts.max_by { |station, count| count }
+    "#{station.first.name} (#{station.last} rides)"
+  end
+
+  def self.station_with_maximum_rides_as_end_location
+    station_counts = group(:end_station).count
+    station        = station_counts.max_by { |station, count| count }
+    "#{station.first.name} (#{station.last} rides)"
+  end
+
+  def self.date_with_highest_number_trips(id)
+    date_counts = where(start_station_id: id).group(:start_date).count("id")
+    date        = date_counts.max_by { |date, count| count }
+    "#{date.first} (#{date.last} rides)"
+  end
+
+  def self.most_frequent_user_zip_code(id)
+    zipcode_counts = where(start_station_id: id).group(:user_zip_code).count("id")
+    zipcode        = zipcode_counts.max_by{|zip_code, count| count }
+    "#{zipcode.first} (#{zipcode.last} rides)"
+  end
+
+  def self.most_frequent_bike_id(id)
+    bike_counts = where(start_station_id: id).group(:bike_id).count("id")
+    bike        = bike_counts.max_by{|bike, count| count }
+    "#{bike.first} (#{bike.last} rides)"
+  end
+
+  ### Trips Dashboard methods ###
+  def self.average_trip_duration
+    average(:duration).to_i / 60
```

Comment:
what does 60 represent? i call numbers like this "magic numbers." meaning - someone without any context reading your code probably has no idea what this represents. i suggest setting this as a constant at the top of this class and referring to it here. that way, it will be much easier to read 👍